test/test-schemas/v2020-12/image.schema.json

```diff
@@ -56,6 +56,17 @@
                 [1.3, 4.03, 42.0],
                 [18.0, 0.1, 1.1]
             ]
+        },
+        "metadata": {
+            "type": "object",
+            "title": "Metadata",
+            "additionalProperties": { "$ref": "meta.schema.json" },
+            "propertyNames": {
+                "enum": ["aperture", "camera", "lens"],
+                "minLength": 3,
+                "maxLength": 10,
+                "pattern": "^[A-Za-z_][A-Za-z0-9_]*$"
+            }
```

Comment:
This is the new feature I wanted to support. I am designing objects in which the values are all the same types but the keys are a specific list. The `propertyNames.enum` pattern perfectly suits this use case. Just need to get it in generated docs.